This short notebook shows how to create a correlation matrix.

In [2]:
import pandas as pd
data = pd.read_csv("QTM_350.csv")
data

Unnamed: 0        Images  Happy  Confused  Calm  Sad  Fear  Angry  \
0            0  IMG_1042.jpg      0        25     0   75     0      0   
1            1  IMG_1043.jpg     25         0    75    0     0      0   
2            2  IMG_1045.jpg      0        25    50   25     0      0   
3            3  IMG_1046.jpg     25         0     0    0    25     25   
4            4  IMG_1047.jpg     25         0     0    0     0      0   
5            5  IMG_1055.jpg      0        25     0   50    25      0   
6            6  IMG_1056.jpg      0        25     0   75     0      0   
7            7  IMG_1057.jpg      0         0     0    0     0     50   
8            8  IMG_1058.jpg     25         0     0    0     0      0   
9            9  IMG_1059.jpg     25        25     0    0     0      0   
10          10  IMG_1060.jpg      0         0     0    0   100      0   
11          11  IMG_1061.jpg      0         0     0    0     0     50   
12          12  IMG_1062.jpg      0         0   100    0     0      0   
13          13  IMG_1063.jpg     25         0     0    0     0      0   
14          14  IMG_1064.jpg      0        25    25   25     0      0   
15          15  IMG_1065.jpg      0         0     0   25    25     25   
16          16  IMG_1066.jpg      0        50    25   25     0      0   
17          17  IMG_1067.jpg      0         0     0    0     0    100   
18          18  IMG_1068.jpg      0         0     0  100     0      0   
19          19  IMG_1069.jpg      0         0     0   25    75      0   

    Surprised  Disgusted  Our correct guess  Rekognition's correct guess  \
0           0          0                 75                        0.235   
1           0          0                 75                       79.900   
2           0          0                 25                        0.782   
3           0         25                  0                       27.107   
4          75          0                 75                       63.805   
5           0          0                  0                        3.511   
6           0          0                 25                       60.974   
7           0         50                 50                        0.758   
8          75          0                 25                        1.179   
9          50          0                  0                       99.500   
10          0          0                100                        5.534   
11          0         50                 50                        1.179   
12          0          0                100                       27.624   
13         75          0                 25                       70.772   
14         25          0                 25                        3.792   
15          0         25                 25                        0.167   
16          0          0                 25                        0.269   
17          0          0                100                       76.237   
18          0          0                100                        7.333   
19          0          0                 75                        0.590   

       Answer  
0         Sad  
1        Calm  
2    Confused  
3    Confused  
4   Surprised  
5   Surprised  
6    Confused  
7   Disgusted  
8       Happy  
9        Calm  
10       Fear  
11  Disgusted  
12       Calm  
13  Surprised  
14        Sad  
15      Angry  
16        Sad  
17      Angry  
18        Sad  
19       Fear

We can check the correlation between our confidence level for emotion and Rekognition's confidence level for the same emotion by first subsetting the two columns.

In [3]:
subset = ["Our correct guess", "Rekognition's correct guess"]
matrix = data[subset]

Now we can make the correlation matrix.

In [4]:
matrix.corr()

Our correct guess  Rekognition's correct guess
Our correct guess                     1.000000                     0.006828
Rekognition's correct guess           0.006828                     1.000000

The correlation between each variable and itself is 1 (as expected), but the correlation between the confidence level for our guess and the confidence level of Rekognition for that same emotion is only 0.0068, implying that there is almost no correlation. 